spring-rabbit/src/main/java/org/springframework/amqp/rabbit/listener/DirectMessageListenerContainer.java

```diff
@@ -355,74 +413,137 @@ protected void actualStart() throws Exception {
 		}
 	}
 
-	private void consumeFromQueue(String queue) throws IOException {
-		for (int i = 0; i < DirectMessageListenerContainer.this.consumersPerQueue; i++) {
+	private void checkMissingQueues(String[] queueNames) {
+		if (isMissingQueuesFatal()) {
+			RabbitAdmin checkAdmin = new RabbitAdmin(getConnectionFactory());
```

Comment:
Can't we use newly introduced `AbstractMessageListenerContainer.getRabbitAdmin()` here ?